epsagon/wrappers/fastapi.py

```diff
@@ -53,24 +61,31 @@ async def custom_route_handler(request: Request) -> Response:
                 collect_container_metadata(runner.resource['metadata'])
             except Exception as exception: # pylint: disable=W0703
                 warnings.warn('Could not extract request', EpsagonWarning)
-
+            raised_err = None
             try:
                 response: Response = await original_route_handler(request)
             except Exception as exception:  # pylint: disable=W0703
+                raised_err = exception
                 traceback_data = get_traceback_data_from_exception(exception)
                 trace.runner.set_exception(exception, traceback_data)
 
-            if response is not None and runner:
-                if ignore_request(
-                        response.headers.get('Content-Type', '').lower(),
-                        ''
-                ):
-                    return response
+            try:
+                if not raised_err and response is not None and runner:
+                    if ignore_request(
+                            response.headers.get('Content-Type', '').lower(),
+                            ''
+                    ):
+                        return response
 
-                runner.update_response(response)
+                    runner.update_response(response)
+
+                if runner:
+                    epsagon.trace.trace_factory.send_traces()
```

Comment:
Where is `send_traces()` done in case that there was a `raised_err`?
If it is not needed maybe we can just raise raised_err in the except block (line #71), instead of adding it to the if and then checking for it and raising later?

Reply:
the send traces will be done even if raised_err :) only the response update won't be performed if raised_err

Reply:
right, my bad :)